### PR TITLE
Refactor Default Colors and the ColorPalette component

### PR DESCRIPTION
**Pull request by youknowriad**
Tiny PR that does two things related to colors:

 - Moves the default colors to the frontend making the Editor script more reusable without the need of the server-side bootstraping.

 - Small fix the ColorPalette component to accommodate the empty colors use-case. (Avoid JS errors)

**Testing instructions**

 - Open Gutenberg
 - Create a paragraph
 - Assign colors

### CODE REVIEW

File: editor/components/provider/index.js [hunk 1]
```diff
@@ -32,6 +32,20 @@ import store from '../../store';
  */
 const DEFAULT_SETTINGS = {
 	alignWide: false,
+	blockTypes: true,
```
Comment (youknowriad) [inline editor/components/provider/index.js]: Same here, this ensure all blocks are available by default.

Comment (aduth) [reply editor/components/provider/index.js]: See #4707. Includes a comment in that PR.

Comment (aduth) [reply editor/components/provider/index.js]: I just merged, so let's drop the change here.

**Review by aduth — DISMISSED** [review]

**Comment by aduth** [comment]
>Moves the default colors to the frontend making the Editor script more reusable without the need of the server-side bootstrapping.

How will we plan to allow plugins to filter this? Or would this need to occur in the client?

**Comment by youknowriad** [comment]
@aduth  Why would we allow plugins to filter this? I mean we can still allow a filter on the frontend but we already have a theme support config to override these colors.

**Comment by aduth** [comment]
The override is a full replacement though, yeah? Maybe not a common use case, but doesn't seem simple as-is for a theme to _add_ a new value.

**Pull request merged**